pilot/pkg/xds/eds.go

```diff
@@ -456,15 +456,18 @@ func shouldUseDeltaEds(req *model.PushRequest) bool {
 // onlyEndpointsChanged checks if a request contains *only* endpoints updates. This allows us to perform more efficient pushes
 // where we only update the endpoints that did change.
 func onlyEndpointsChanged(req *model.PushRequest) bool {
-	if len(req.ConfigsUpdated) > 0 {
-		for k := range req.ConfigsUpdated {
-			if k.Kind != kind.ServiceEntry {
-				return false
-			}
+	if len(req.ConfigsUpdated) == 0 {
+		return false
+	}
+	for cfg := range req.ConfigsUpdated {
+		if _, f := skippedEdsConfigs[cfg.Kind]; f {
```

Comment:
A comment here would help a lot - not sure I follow the logic, so needs comments. 

Before: return false if kind == ServiceEntry, return true otherwise.
After: return false if kind == ServiceEntry (next check), return true otherwise ( continue moves to the next).

Reply:
OK, added some comments